### PR TITLE
protocols/gossipsub: Fix duplicate connection id

### PR DESCRIPTION
**Pull request by ackintosh**
# Description

At the first connection for a peer, duplicated `connection_id` are inserted into the `PeerConnections::connections` field.

## Change checklist

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

**Review by divagant-martian — APPROVED** [review]
Nice, thanks

**Review by mxinden — COMMENTED** [review]
Good catch. Thanks @ackintosh. Mind bumping the `libp2p-gossipsub` patch version and adding a changelog entry?

Does this result in any downstream issues? If so, I am happy to cut a patch release right away.

**Comment by ackintosh** [comment]
@mxinden Thank you for your review. Bumped up the patch version and added a changelog entry.

> Does this result in any downstream issues?

I think yes. If there are two or more actual connections between a peer, and when one of them is closed, [this `connections[0]`](https://github.com/libp2p/rust-libp2p/blob/5cbd446b264ba4188b4b799a4dcfe2e8c7be90cb/protocols/gossipsub/src/behaviour.rs#L3197) may point to the connection id already closed. This results in a lost connection to the mesh. (I have not seen the issue occur though because I noticed this issue while just I was reading the implementation.)

**Comment by divagant-martian** [comment]
@mxinden no rush on our side. I checked and this code has been there for quite some time

File: protocols/gossipsub/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+# 0.38.1 [unreleased]
+
+- Fix duplicate connection id. See [PR 2702]
```
Comment (mxinden) [inline protocols/gossipsub/CHANGELOG.md]:
```suggestion
# 0.38.1

- Fix duplicate connection id. See [PR 2702].
```

I will do a release right away.

**Review by mxinden — APPROVED** [review]

**Comment by mxinden** [comment]
[Tagged](https://github.com/libp2p/rust-libp2p/releases/tag/libp2p-gossipsub-0.38.1) and [published](https://crates.io/crates/libp2p-gossipsub/0.38.1).

**Comment by ackintosh** [comment]
Thank you @mxinden 😃

**Pull request merged**